### ISSUE TITLE
[routing] Run out of memory (32-bit systems)

### ISSUE THREAD

**Issue by vng** [organicmaps/organicmaps#2296]
Download next regions:
Rheinland-Pfalz - KOBLENZ
Rheinland-Pfalz - MAINZ
Hessen -> DARMSTADT
Baden-Würtemberg -> KARLSRUHE
Baden-Würtemberg -> STUTTGART-HEILBRONN
Baden-Würtemberg -> STUTTGART- STUTTGART

Build **bicycle** route Koblenz -> Stuttgart

iPhone 6 (or similar arm32 device) consumes 700 Mb RAM and the app is killed.

**Comment by biodranik** [comment]
No, it's not related to 32bits. My iPad is 64 bits: https://www.gsmarena.com/apple_ipad_mini_3-6741.php

**Comment by vng** [comment]
Yes. It also happens on arm64, the app memory is limited by 2Gb or so.
Bicycle and Pedestrian routing work in a different way, rather than Car. And they are not suitable for the long distance route building in a dense areas across many MWMs.

Probably, we should invent some _leaps mode_ for them, like in a Car routing.
